stem/output.py

```diff
@@ -0,0 +1,219 @@
+        """
+        Abstract method for validating user inputs
+        """
+        pass
+
+
+@dataclass
+class GiDOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for GiD output
+
+    Attributes:
+        gidpost_flags (Dict[str, Any]):
+        file_label (str):
+        output_control_type (str):
+        output_interval (int):
+        body_output (bool):
+        node_output (bool):
+        skin_output (bool):
+        plane_output (List[str]):
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        point_data_configuration (List[str]):
+    """
+
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+
+    gid_post_mode: str = "GiD_PostBinary"
+    file_label: str = "step"
+    output_control_type: str = "step"
+    output_interval: int = 1
+    body_output: bool = True
+    node_output: bool = False
+    skin_output: bool = False
+    # optional
+    plane_output: List[str] = field(default_factory=lambda: [])
+    point_data_configuration: List[str] = field(default_factory=lambda: [])
+
+    def validate(self):
+        pass
+
+
+@dataclass
+class VtkOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for GiD output
+
+    Attributes:
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        file_format (str): file format for VTK, either `binary` or `ascii` are allowed.
+        output_precision (int):
+        output_control_type (str): type of output control, either `step` or `time`.
+        output_interval (float):
+    """
+    # general inputs
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+    # VTK specif inputs
+    file_format: str = "binary"
+    output_precision: int = 7
+    output_control_type: str = "step"
+    output_interval: float = 1.0
+
+    def validate(self):
+        detect_tensor_outputs(requested_outputs=self.gauss_point_results)
+
+
+@dataclass
+class JsonOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for JSON output
+
+    Attributes:
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        time_frequency (float):
+        sub_model_part_name (str):
+    """
+    # general inputs
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+    # JSON specif inputs
+    time_frequency: float = 1.0
+    sub_model_part_name: str = ""
+
+    def validate(self):
+        detect_tensor_outputs(requested_outputs=self.gauss_point_results)
+
+
+class Output:
+    """
+    Class containing output information for postprocessing
+
+    Attributes:
+        part_name (str): name of the model part
+        output_path (str): output path for the output
```

Comment:
I'm rethinking output_path. Maybe this can be split in output_dir and output_name, where output_name is only used in gid and json. And then in the KratosIO output_dir and output_name are put together for the projectparameters file. 

If the output_name is not given, the model_part name should be used as output name. 

I think this is a bit more clear, what do you think?

Reply:
Solved in the same way. We need to keep in mind a few things:
- for outputs JSON the directory has to exist or it will fail. Implementation of path.mkdirs should be adopted but now skipped because of testing.
- For VTK and GiD the directory doesn't need to exist as KRATOS creates it.